Src/AutoNSubstituteUnitTest/Scenario.cs

```diff
@@ -0,0 +1,97 @@
+﻿using NSubstitute;
+using NSubstitute.Exceptions;
+using Ploeh.AutoFixture.Xunit;
+using Xunit;
+using Xunit.Extensions;
+
+namespace Ploeh.AutoFixture.AutoNSubstitute.UnitTest
+{
+    public class Scenario
+    {
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void InterfacesAreSubstitutedByDefault(IInterface i, object obj)
+        {
+            i.Method(obj);
+            i.Received().Method(obj);
+        }
+
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void AbstractTypesAreSubstitutedByDefault(AbstractType t, object obj)
+        {
+            t.AbstractMethod(obj);
+            t.Received().AbstractMethod(obj);
+        }
+
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void ConcreteTypesAreNotSubstitutedByDefault(ConcreteType t, object obj)
+        {
+            t.VirtualMethod(obj);
+            Assert.Throws<NotASubstituteException>(() => t.Received().VirtualMethod(obj));
+        }
+
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void ConcreteTypesAreSubstitutedWhenParameterIsMarkedWithSubstituteAttribute(
+            [Substitute]ConcreteType t, object obj)

```

Comment:
Now that I see what this looks like, I wonder why, instead of an attribute, we don't introduce a `Substitute<T>` class?

As we've previously discussed, the order of attributes isn't guaranteed by the .NET framework, so when `[Substitute]` is an attribute, you can't combine it with e.g. `[Frozen]`.

IMO, it wouldn't hurt readability, and it's the same number of characters you'd have to type:

```C#
[Theory, AutoDataWithAutoSubstituteCustomization]
public void ConcreteTypesAreSubstitutedWhenParameterIsMarkedWithSubstituteAttribute(
    Substitute<ConcreteType> t, object obj)
```

That would solve the combination problem.